nixos/lib/test-driver/test_driver/machine.py

```diff
@@ -720,6 +727,77 @@ def screenshot(self, filename: str) -> None:
             if ret.returncode != 0:
                 raise Exception("Cannot convert screenshot")
 
+    @contextmanager
+    def record_audio(
+        self,
+        filename: str,
+        allow_silence: bool = False,
+        device: str = "audio_device",
+        sample_rate: int = 44100,
+        bit_depth: int = 16,
+        channels: int = 2,
+    ) -> Iterator[None]:
+        filename = add_extension_if_simple_name(filename, "wav")
```

Comment:
what do you think about simply enforcing that the user provided a sane filename and throwing a nice self-explaining exception otherwise?

Reply:
Oh I think something got lost in the rebase, but this is the same logic we use for screenshots. I factored out the function, but didn't use it for screenshot